### ISSUE TITLE
NESTED: using a nested GET api/serializer writes only headers, fields are empty - fix in PR #36

### ISSUE THREAD

**Issue by ionescu77** [wharton/drf-renderer-xlsx#37]
there is already a fix, a PR (#36) opened by @paveloder 

I have applied the fix locally and I can confirm it works.

My API was returning:

```json
HTTP 200 OK
Allow: GET
Content-Type: application/json
Vary: Accept

[
    {
        "id": 1,
        "rfid_tag": {
            "id": 1,
            "number": "1234567890",
            "product": {
                "id": 1,
                "name": "Prosop mare",
                "ean": "11111",
                "description": "",
                "category": 1,
                "info": []
            },
            "status": "I",
            "info": []
        },
        "security_location": 1,
        "aknowledged": false
    },
    {
        "id": 2,
        "rfid_tag": {
            "id": 2,
            "number": "1234567891",
            "product": {
                "id": 1,
                "name": "Prosop mare",
                "ean": "11111",
                "description": "",
                "category": 1,
                "info": []
            },
            "status": "I",
            "info": []
        },
        "security_location": 2,
        "aknowledged": false
    }
]
```

- viewset

```py
class SecurityEventsGetXLSXViewset(XLSXFileMixin, ReadOnlyModelViewSet):
    """ Generate XLSX File """
    queryset = models.SecurityEvents.objects.all()
    serializer_class = serializers.SecurityEventsGetSerializer
    renderer_classes = (XLSXRenderer,)
    filename = 'SecurityEventsGet_export.xlsx'
```

- serializer
```py
class SecurityEventsGetSerializer(serializers.ModelSerializer):
    rfid_tag = RfidTagNestedSerializer()
    class Meta:
        model = models.SecurityEvents
        fields = ['id', 'rfid_tag', 'security_location', 'aknowledged']
        read_only_fields = ['created_at']
```
## Before PR #36 patch

<img width="854" alt="Screenshot 2021-06-22 at 15 02 21" src="https://user-images.githubusercontent.com/8016748/122921663-67b7eb00-d36b-11eb-9c26-9d1e3be32b1f.png">

## After PR #36 patch

<img width="861" alt="Screenshot 2021-06-22 at 15 02 30" src="https://user-images.githubusercontent.com/8016748/122921519-422ae180-d36b-11eb-905e-d9765eea5a01.png">

**Comment by FlipperPA** [comment]
@ionescu77 I've merged in pull requests #35 and #36, which created a conflict. I think I've sorted out both properly, but could you please run a test against the latest on the `main` branch before I push out a release and confirm that it works?

**Comment by ionescu77** [comment]
Hello, I will give it a try and report back. Thank you!

**Comment by ionescu77** [comment]
It's been a while. Thanks for your patience.

I tried local with the `renderers.py` from `main` branch. It does not seem to work as expected.

Using the example from 14 days ago I get something like

<img width="812" alt="Screenshot 2021-07-06 at 16 02 42" src="https://user-images.githubusercontent.com/8016748/124604425-9a85d700-de73-11eb-9b77-3248e5e32d60.png">



 Let me try to look at #35  #36  see if I spot the trouble.

**Comment by ionescu77** [comment]
Excuse all, I deleted previous comment (it was my problem, I made some changes to `serializers.py` in my project and it impacted XLSX generation, too).

## Test results

So:
- [ ] I have copied `renderers.py` from `main` branch into my project venv `lib/python3.9/srf_renderer_xlsx`.
- [ ] I used another nested serializer in order to test (actually multiple level nesting: `RfidTag` -> `Product` -> `Category`)
- [ ] It seems I get my XLSX file as expected (nested product info is added as columns next to RFID Tag epc number):
- [ ] congrats for the PR merge and thank you! 🏆

```py
# api_views.py
class SecurityEventsGetXLSXViewset(XLSXFileMixin, ReadOnlyModelViewSet):
    """ Generate XLSX File """
    queryset = models.RfidTag.objects.all()
    serializer_class = serializers.RfidTagNestedSerializer
    renderer_classes = (XLSXRenderer,)
    filename = 'RfidTagNested_export.xlsx'
```

```py
# serializers.py
class RfidTagNestedSerializer(serializers.ModelSerializer):
    product = ProductNestedSerializer()
    class Meta:
        model = models.RfidTag
        fields = ['id', 'number', 'product', 'state', 'info']
```

```json
[
    {
        "id": 1,
        "number": "111abc",
        "product": {
            "id": 1,
            "name": "default_product",
            "ean": "",
            "description": "default product for foreign/unknown RFID tags",
            "category": {
                "id": 1,
                "name": "default_category",
                "description": "default category for foreign/unknown RFID tags"
            },
            "info": []
        },
        "state": "",
        "info": []
    },
    {
        "id": 2,
        "number": "111ABC",
        "product": {
            "id": 1,
            "name": "default_product",
            "ean": "",
            "description": "default product for foreign/unknown RFID tags",
            "category": {
                "id": 1,
                "name": "default_category",
                "description": "default category for foreign/unknown RFID tags"
            },
            "info": []
        },
        "state": "",
        "info": []
    },
    {
        "id": 3,
        "number": "111ABCD",
        "product": {
            "id": 1,
            "name": "default_product",
            "ean": "",
            "description": "default product for foreign/unknown RFID tags",
            "category": {
                "id": 1,
                "name": "default_category",
                "description": "default category for foreign/unknown RFID tags"
            },
            "info": []
        },
        "state": "",
        "info": []
    },
    {
        "id": 4,
        "number": "111ABCDE",
        "product": {
            "id": 3,
            "name": "Prosop mic",
            "ean": "222",
            "description": "",
            "category": {
                "id": 2,
                "name": "Textile Camera",
                "description": ""
            },
            "info": []
        },
        "state": "I",
        "info": []
    },
    {
        "id": 5,
        "number": "111ABCDEF",
        "product": {
            "id": 2,
            "name": "Prosop mare",
            "ean": "111",
            "description": "",
            "category": {
                "id": 2,
                "name": "Textile Camera",
                "description": ""
            },
            "info": []
        },
        "state": "I",
        "info": []
    },
    {
        "id": 6,
        "number": "string",
        "product": {
            "id": 1,
            "name": "default_product",
            "ean": "",
            "description": "default product for foreign/unknown RFID tags",
            "category": {
                "id": 1,
                "name": "default_category",
                "description": "default category for foreign/unknown RFID tags"
            },
            "info": []
        },
        "state": "",
        "info": []
    },
    {
        "id": 7,
        "number": "abc",
        "product": {
            "id": 1,
            "name": "default_product",
            "ean": "",
            "description": "default product for foreign/unknown RFID tags",
            "category": {
                "id": 1,
                "name": "default_category",
                "description": "default category for foreign/unknown RFID tags"
            },
            "info": []
        },
        "state": "",
        "info": []
    },
    {
        "id": 8,
        "number": "3034802c801c4ad6a5a4372e",
        "product": {
            "id": 1,
            "name": "default_product",
            "ean": "",
            "description": "default product for foreign/unknown RFID tags",
            "category": {
                "id": 1,
                "name": "default_category",
                "description": "default category for foreign/unknown RFID tags"
            },
            "info": []
        },
        "state": "",
        "info": []
    }
]
```

And the XLSX:

<img width="959" alt="Screenshot 2021-07-06 at 17 00 26" src="https://user-images.githubusercontent.com/8016748/124613219-abd2e180-de7b-11eb-8d7b-2cb06721b737.png">

**Comment by FlipperPA** [comment]
The fix has been released in version 0.4.1: https://pypi.org/project/drf-renderer-xlsx/0.4.1/
